remove caching of macro names and values in glslc

Qining opened the pull request.

dneto0 commented on glslc/src/main.cc:
```diff
@@ -212,23 +205,15 @@ int main(int argc, char** argv) {
-        macro_values.emplace_back(
-            equal_sign_loc == shaderc_util::string_piece::npos ||
+        const string_piece value_piece =
+            equal_sign_loc == string_piece::npos ||
```
Please put parentheses around the expression before the ?
Makes it clearer and idiot-proof.

Qining replied:
Yes, I should have it in parentheses before. Now it looks better.

dneto0 commented:
-1 for a nit on parenthesizing the expression.

dneto0 commented:
Can you explain the Windows Release failure in AppVeyor?
Otherwise looks ok.

Qining commented:
I added the parentheses at a wrong place, changed the '=' logical and collapsed the whole macro_value string_piece.

Should have already been fixed.

dneto0 commented:
+2

dneto0 commented:
Rebased and pushed into master as e145d34a0
